src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -2647,16 +2648,40 @@ public RegistryAuth registryAuth() {
      *
      * @param registryAuth RegistryAuth object
      * @return Builder
+     *
+     * @deprecated in favor of {@link #registryAuthSupplier(RegistryAuthSupplier)}
      */
+    @Deprecated
     public Builder registryAuth(final RegistryAuth registryAuth) {
+      if (this.registryAuthSupplier != null) {
```

Comment:
I'd like to remove all the deprecated methods in a future release - see what I added to Changelog